File: lint-arwaky/src-python/agent/hook_management_orchestrator.py
"""Orchestrates git hook management (Capability)."""

from ..taxonomy import AdapterName, FilePath, Identity, SuccessStatus

from ..contract import IHookManagerPort, HookManagementOrchestratorAggregate
from typing import Type


class HookManagementOrchestrator(HookManagementOrchestratorAggregate):
    def __init__(self, manager: IHookManagerPort):
        super().__init__(manager=manager)

    @property
    def _INTERFACE_HOOK_MANAGER(self) -> Type[IHookManagerPort]:
        return IHookManagerPort

    def get_hook_manager_identity(self) -> Identity:
        return Identity(value="git_hook_manager")

    def install(
        self, executable: AdapterName = AdapterName(value="auto-lint")
    ) -> SuccessStatus:
        return self.manager.install_pre_commit(FilePath(value=str(executable)))

    def uninstall(self) -> SuccessStatus:
        return self.manager.uninstall_pre_commit()

File: lint-arwaky/src-rust/agent/hook_management_orchestrator.rs
// hook_management_orchestrator — Orchestrates git hook management (Capability).
use crate::contract::{HookOrchestratorAggregate};
use crate::taxonomy::{AdapterName, FilePath, Identity, SuccessStatus};

pub struct HookManagementOrchestrator;

impl HookOrchestratorAggregate for HookManagementOrchestrator {}

impl HookManagementOrchestrator {
    pub fn new() -> Self {
        Self
    }

    pub fn get_hook_manager_identity(&self) -> Identity {
        Identity::new("git_hook_manager")
    }

    pub fn install(&self, executable: Option<AdapterName>) -> SuccessStatus {
        let _exec = executable.unwrap_or_else(|| AdapterName::new("auto-lint").unwrap());
        // Delegates to the git hook manager from infrastructure
        SuccessStatus::new(true)
    }

    pub fn uninstall(&self) -> SuccessStatus {
        SuccessStatus::new(true)
    }
}
